src/BloomBrowserUI/bookEdit/js/readerTools.js

```diff
@@ -2,78 +2,122 @@
             return;
-
         case 'Words':
             var words = ReaderToolsModel.selectWordsFromSynphony(false, params[1].split(' '), params[1].split(' '), true, true);
             getSetupDialogWindow().postMessage('Words\n' + JSON.stringify(words), '*');
             return;
-
         case 'Refresh':
             var synphony = model.getSynphony();
             synphony.loadSettings(JSON.parse(params[1]));
             model.updateControlContents();
             model.doMarkup();
             return;
-
         case 'SetupType':
             getSetupDialogWindow().postMessage('SetupType\n' + model.setupType, '*');
             return;
-
         case 'SetMarkupType':
             model.setMarkupType(parseInt(params[1]));
             return;
-
+        case 'Qtips':
+            // We could make separate messages for these...
+            markDecodableStatus();
+            markLeveledStatus();
+            return;
         default:
     }
 }
-
+function markDecodableStatus() {
+    // q-tips; mark sight words and non-decodable words
+    var editableElements = $(".bloom-content1");
+    editableElements.find('span.' + $.cssSightWord()).each(function () {
+        $(this).qtip({ content: 'Sight word' });
+    });
+    editableElements.find('span.' + $.cssWordNotFound()).each(function () {
+        $(this).qtip({ content: 'This word is not decodable in this stage.' });
+    });
+    // we're considering dropping this entirely
+    // We are disabling the "Possible Word" feature at this time.
+    //editableElements.find('span.' + $.cssPossibleWord()).each(function() {
+    //    (<qtipInterface>$(this)).qtip({ content: 'This word is decodable in this stage, but is not part of the collected list of words.' });
+    //});
+}
+function markLeveledStatus() {
+    // q-tips; mark sentences that are too long
+    var editableElements = $(".bloom-content1");
+    editableElements.find('span.' + $.cssSentenceTooLong()).each(function () {
+        $(this).qtip({ content: 'This sentence is too long for this level.' });
+    });
+}
+function getReaderToolsModel() {
+    var accordion = parent.window.document.getElementById("accordion");
+    // accordion will be undefined during unit testing
+    if (accordion)
+        return accordion.contentWindow['model'];
+}
+function setupReaderKeyAndFocusHandlers(container, model) {
+    // invoke function when a bloom-editable element loses focus.
+    $(container).find('.bloom-editable').focusout(function () {
+        model.doMarkup();
+    });
+    $(container).find('.bloom-editable').focusin(function () {
+        model.noteFocus(this); // 'This' is the element that just got focus.
+    });
+    // and a slightly different one for keypresses
+    $(container).find('.bloom-editable').keypress(function () {
+        model.doKeypressMarkup();
+    });
+    $(container).find('.bloom-editable').keydown(function (e) {
```

Comment:
Did we lose a comment somehow about why we need to do this rather obscure thing? (If not it's probably one I should have written, sorry.)